ios/FluentUI/Popup Menu/PopupMenuProtocols.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+/// Defines the timing for the call of the onSelected closure/block
+@objc(MSFPopupMenuItemExecutionMode)
+public enum ExecutionMode: Int {
+    /// `onSelected` is called right after item is tapped, before popup menu dismissal
+    case onSelection
+    /// `onSelected` is called right after item is tapped, but prevent popup menu dismissal
+    case onSelectionWithoutDismissal
+    /// `onSelected` is called after popup menu is dismissed, but before its `onDismissCompleted` is called
+    case afterPopupMenuDismissal
+    /// `onSelected` is called after popup menu is dismissed and its `onDismissCompleted` is called
+    case afterPopupMenuDismissalCompleted
+}
+
+/**
+`PopupMenuTemplateItem` represents a template item protocol inside `PopupMenuController`.
+ The built-in type is `PopupMenuItem`.
+ You can use object conforms to this protocol for customization.
+*/
+@objc(MSFPopupMenuTemplateItem)
+public protocol PopupMenuTemplateItem: AnyObject {
+    /// The custom cell class for `PopupMenuController`
+    @objc var cellClass: PopupMenuItemTemplateCell.Type { get set}
+    @objc var executionMode: ExecutionMode { get }
+    @objc var isSelected: Bool { get set }
+    @objc var onSelected: (() -> Void)? { get }
+}
+
+/**
+`PopupMenuItemTemplateCell` represents a template cell protocol inside `PopupMenuController`.
+ The built-in type is `PopupMenuItemCell`.
+ You can use `UITableViewCell` conforms to this protocol for customization.
+*/
+@objc(MSFPopupMenuItemTemplateCell)
+public protocol PopupMenuItemTemplateCell where Self: UITableViewCell {
+    /// `PopupMenuController` will notify that one or more items in the list contain image(s)
+    @objc var preservesSpaceForImage: Bool { get set }
+    /// `PopupMenuController` will notify the expected separatorColor.
+    /// For custom cell, you should add your own separator.
+    @objc var expectedSeparatorColor: UIColor? { get set }
```

Comment:
i think it should be customSeparatorColor

Reply:
Done.